BeeSwift/GoalViewController.swift

```diff
@@ -24,6 +24,14 @@ class GoalViewController: UIViewController,  UIScrollViewDelegate, DatapointTabl
 
     let goal: Goal
 
+
+    private var lastFetched: Date? {
```

Comment:
This doesn't seem to be a real variable. By this I mean we don't really keep track of a value in it - the only reads are within `updateLastUpdatedLabel`, which could just as happily be reading directly from `goal.lastModifiedLocal`

[Do] Let's remove it, and just call `updateLastUpdatedLabel()` in the places we were writing to it.

Reply:
done

Reply:
unfortunately that has the side effect of the app crashing when accessing goal.lastModifiedLocal at GoalViewController.updateLastUpdatedLabel upon signing into the app.

1. GoalManager.performPostGoalUpdateBookkeeping posts notification goalsUpdated
2. GoalViewController.onGoalsUpdatedNotification executes
3. calling its updateInterfaceToMatchGoal
4. GoalViewController.updateInterfaceToMatchGoal calls, among others, updateLastUpdatedLabel
5. GoalViewController.updateLastUpdateLabel now accesses self.goal.lastModifiedLocal , which at the time, is still part of an empty NSManagedObject Goal:

```swift
<Goal: 0x600002148aa0> (entity: Goal; id: 0xblahblahblah9494 <x-coredata://blahblah-F9F9-4FB3-BD1B-blahblah/Goal/p385>; data: {
    alertStart = nil;
    autodata = nil;
    data = "<relationship fault: 0x600000383740 'data'>";
    deadline = nil;
    graphUrl = nil;
    healthKitMetric = nil;
    hhmmFormat = nil;
    id = nil;
    initDay = nil;
    lastModifiedLocal = nil;
    lastTouch = nil;
    leadTime = nil;
    limSum = nil;
    owner = nil;
    pledge = nil;
    queued = nil;
    recentData = "<relationship fault: 0x6000003829a0 'recentData'>";
    safeBuf = nil;
    safeSum = nil;
    slug = nil;
    thumbUrl = nil;
    title = nil;
    todayta = nil;
    urgencyKey = nil;
    useDefaults = nil;
    won = nil;
    yAxis = nil;
})
```

Reply:
It seems likely that versions of this bug will end up existing, or be easy to introduce, all across GoalView. Do you have ideas for why the GoalViewController even exists while in this state? We're certainly not showing it.

Reply:
Tracking this as #590